File: MultiLanguageSandbox/src/test/deps/C#/33.cs
using System;
using System.Collections.Generic;
using System;
using System.Collections.Generic;
using System.Diagnostics;

class Program
{

/* Finds all automorphic numbers within a specified limit.
    An automorphic number is a number whose square ends in the number itself.
    Examples:
    >>> FindAutomorphicNumbers(100)
    [0, 1, 5, 6, 25, 76]
    >>> FindAutomorphicNumbers(500)
    [0, 1, 5, 6, 25, 76, 376]
*/

static List<int> FindAutomorphicNumbers(int limit)
{
        List<int> automorphicNumbers = new List<int>();
        
        for (int i = 0; i <= limit; i++)
        {
            long square = (long)i * i; // Use long to avoid overflow for larger numbers
            string numberStr = i.ToString();
            string squareStr = square.ToString();
            
            if (squareStr.EndsWith(numberStr))
            {
                automorphicNumbers.Add(i);
            }
        }
        
        return automorphicNumbers;
    }
    static void Main()
    {
     Debug.Assert(FindAutomorphicNumbers(100).SequenceEqual(new List<int> { 0, 1, 5, 6, 25, 76 }));
    Debug.Assert(FindAutomorphicNumbers(500).SequenceEqual(new List<int> { 0, 1, 5, 6, 25, 76, 376 }));
    Debug.Assert(FindAutomorphicNumbers(1000).SequenceEqual(new List<int> { 0, 1, 5, 6, 25, 76, 376, 625 }));
    Debug.Assert(FindAutomorphicNumbers(200000).SequenceEqual(new List<int> { 0, 1, 5, 6, 25, 76, 376, 625,9376, 90625, 109376 }));

    }
}

File: MultiLanguageSandbox/src/test/deps/Visual Basic/33.vb
Module Module1
Function FindAutomorphicNumbers(ByVal start As Integer, ByVal endVal As Integer) As List(Of Integer)
    Dim automorphicNumbers As New List(Of Integer)
    
    For i As Integer = start To endVal
        Dim square As Long = CLng(i) * CLng(i) ' Use Long to avoid overflow for larger numbers
        Dim strNumber As String = i.ToString()
        Dim strSquare As String = square.ToString()
        
        ' Check if the square ends with the original number
        If strSquare.EndsWith(strNumber) Then
            automorphicNumbers.Add(i)
        End If
    Next
    
    Return automorphicNumbers
End Function

    Sub CheckAutomorphic(ByVal testFunction As Func(Of Integer, Integer, List(Of Integer)))
        ' Test cases
        Debug.Assert(testFunction(0, 10).SequenceEqual(New List(Of Integer) From {0, 1, 5, 6}))
        Debug.Assert(testFunction(25, 30).SequenceEqual(New List(Of Integer) From {25}))
        Debug.Assert(testFunction(75, 100).SequenceEqual(New List(Of Integer) From {76}))

        Console.WriteLine("FindAutomorphicNumbers function passed all tests")
    End Sub

    Sub Main()
        ' Testing the FindAutomorphicNumbers function
        CheckAutomorphic(AddressOf FindAutomorphicNumbers)
    End Sub
End Module